### PR TITLE
feat(lltz_michelson): Conversion of primitives

### PR DESCRIPTION
**Pull request by alanmarkoTrilitech**
# Context
https://app.asana.com/0/1207308081067689/1208023639799545

# Description
This PR compilies primitives from the LLTZ-IR to Michelson AST. Failwith for a few where LLTZ types don't match (will be fixed in later PRs).

# Manual testing
```dune build ```

### CODE REVIEW

File: lib/lltz_michelson/lltz_michelson.ml [hunk 1]
```diff
@@ -87,4 +88,96 @@ and convert_constant (const: LLTZ.E.constant) : Michelson.Ast.t =
     | Bls12_381_g1 _ -> Michelson.T.bls12_381_g1
     | Bls12_381_g2 _ -> Michelson.T.bls12_381_g2
     | Bls12_381_fr _ -> Michelson.T.bls12_381_fr
-    | Signature _ -> Michelson.T.signature
\ No newline at end of file
+    | Signature _ -> Michelson.T.signature
+
+let convert_primitive (prim: LLTZ.P.t) : Michelson.Ast.t =
+  match prim with
```
Comment (rinderknecht) [inline lib/lltz_michelson/lltz_michelson.ml:94]: This is an exception to the rule "No (global) openings of modules.": ASTs. Use here an `let open Michelson.Ast.Instruction in`.

Comment (alanmarkoTrilitech) [reply lib/lltz_michelson/lltz_michelson.ml]: added ✅

File: lib/lltz_michelson/lltz_michelson.ml [hunk 2]
```diff
@@ -87,4 +88,97 @@ and convert_constant (const: LLTZ.E.constant) : Michelson.Ast.t =
...
+
+let convert_primitive (prim: LLTZ.P.t) : Michelson.Ast.t =
+  let open Michelson.Ast.Instruction in
+    match prim with
+    | Amount -> amount
+    | Balance -> balance
+    | Chain_id -> chain_id
+    | Level -> level
+    | Now -> now
+    | Self opt -> (match opt with Some str -> self | None -> self)
```
Comment (rinderknecht) [inline lib/lltz_michelson/lltz_michelson.ml:101]: | Self Some str -> self
| Self None -> self

But don't you have a warning for the unused variable `str`?

Comment (alanmarkoTrilitech) [reply lib/lltz_michelson/lltz_michelson.ml:101]:
```suggestion
    | Self opt -> failwith
```

Comment (alanmarkoTrilitech) [reply lib/lltz_michelson/lltz_michelson.ml]: right, changed it to failwith for now as its another case with these string options which seem wrongly defined in the ir.

**Review by rinderknecht — APPROVED** [review]

**Pull request merged**